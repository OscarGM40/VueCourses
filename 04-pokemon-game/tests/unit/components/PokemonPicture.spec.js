import { shallowMount } from "@vue/test-utils";

import PokemonPicture from "@/components/PokemonPicture";

describe("Test suite(PokemonPicture)", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("1- should match snapshot", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 1,
        showPokemon: false,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("2- should show hidden img and pokemon with id of 100", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: false,
      },
    });
    // find traerá la primera,me vale
    const img = wrapper.find("img");
    // podria asegurarme que la prop pasa el valor
    expect(wrapper.props("pokemonId")).toBe(100);

    // pero aún mejor sería afirmar que la img ha seteado su atributo src correctamente con ese id.Para ello tenemos el método attributes() que devuelve los atributos de una tag(muy parecido a props())
    expect(img.attributes("src")).toBe(
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg",
    );

    // para las clases podia haber usado atrributes() pero también tenemos cmp.classes que trae un array con las clases.Dado que es un array puedo hacer toContain para buscar por un elemento en concreto
    expect(img.classes()).toContain("hidden-pokemon");
    // otra forma sería asi
    expect(img.classes("hidden-pokemon")).toBe(true);
  });

  test("3- should show pokemon with full brightness if showPokemon comes as true", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 55,
        showPokemon: true,
      },
    });
    const [img1, img2] = wrapper.findAll("img");
    expect(img1.exists()).toBe(true);
    expect(img2).toBe(undefined);
    expect(img1.classes("fade-in")).toBe(true);
  });
});
