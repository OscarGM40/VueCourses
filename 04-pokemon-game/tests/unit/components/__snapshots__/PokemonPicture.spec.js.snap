// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite(PokemonPicture) 1- should match snapshot 1`] = `<div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" class="hidden-pokemon" alt="pokemon"></div>`;
